### PR TITLE
Netfx net standard support package for net462/net47

### LINKED ISSUE #18567: Add .NET Framework 4.7 support to NETStandard.Library.NETFramework package
We need a separate build of both ref (shims.proj so that it unifies with ValueTuple in mscorlib) and lib to support net47. Without that we'd have type-conflicts at compile-time. 

cc @ericstj

### PR DESCRIPTION
**Pull request by ericstj**
This adds support for version-specific ref and implementation in our desktop support package.  Right now we have 1 file that differs (ValueTuple on net47) but I suspect there are some facades that could be fixed in order to better fill in the existing netstandard API on net462 (Crypto).

@weshaggard I'd like your help identifying those by getting API compat runs for net462 and net47 if possible.

Please review commit-by-commit.  More detail about each commit is in the description.

Fixes #18567

### CODE REVIEW

File: Tools-Override/FrameworkTargeting.targets [hunk 1]
```diff
@@ -263,8 +263,8 @@
     </ItemGroup>
 
     <PropertyGroup>
-      <_propsFilename>$(TargetName)</_propsFilename>
-      <_propsFilename Condition="'$(_propsFilename)' == ''">$(MSBuildProjectName)</_propsFilename>
+      <_propsFilename>$(TargetName).$(TargetGroup)-$(OSGroup)</_propsFilename>
```
Comment (weshaggard) [inline Tools-Override/FrameworkTargeting.targets]: Why not Configuration? That allows someone to build both Debug and Release as well (or you can do that at a higher level).

File: dir.targets [hunk 2]
```diff
@@ -63,12 +63,13 @@
       <!-- enable trimming for any runtime project that's part of the shared framework and hasn't already set ILLinkTrimAssembly -->
       <SetProperties Condition="'$(BinPlaceRuntime)' == 'true' AND '$(ILLinkTrimAssembly)' == ''">ILLinkTrimAssembly=true</SetProperties>
     </BinPlaceConfiguration>
+    <BinPlaceConfiguration Condition="'$(IsUAPRef)'=='true' AND ('$(BuildingUAPVertical)' == 'true' OR '$(BuildingUAPAOTVertical)' == 'true')" Include="uap-$(_bc_OSGroup)">
```
Comment (weshaggard) [inline dir.targets]: Do we have an uapaot configurations to clean out of ref configs?

**Comment by weshaggard** [comment]
>@weshaggard I'd like your help identifying those by getting API compat runs for net462 and net47 if possible.

Once this change is in I can add some more apicompat runs for those configs.

**Comment by bartonjs** [comment]
System.Security.Cryptography.Algorithms has [System.Security.Cryptography.ECCurve](https://msdn.microsoft.com/en-us/library/system.security.cryptography.eccurve(v=vs.110).aspx), which is new in net47, so that library might need a new facade to get the new type forwards.

File: dir.targets (hunk 2 above)
Comment (ericstj) [inline dir.targets]: Nope.  I did a scrub before removing this.

**Pull request merged**